tst/EnergyPlus/unit/FaultsManager.unit.cc

```diff
@@ -312,4 +315,142 @@ TEST_F(EnergyPlusFixture, FaultsManager_CalFaultOffsetAct)
+
+        "  FaultModel:HumiditySensorOffset:OutdoorAir,",
+        "    OAWFault,                !- Name",
+        "    ALWAYS_ON,               !- Availability Schedule Name",
+        "    ,                        !- Severity Schedule Name",
+        "    Controller:OutdoorAir,   !- Controller Object Type",
+        "    VAV_1_OA_Controller,     !- Controller Object Name",
+        "    -0.002;                  !- Humidity Sensor Offset {kgWater/kgDryAir}",
+
+        "  FaultModel:EnthalpySensorOffset:OutdoorAir,",
+        "    OAHFault,                !- Name",
+        "    ALWAYS_ON,               !- Availability Schedule Name",
+        "    ,                        !- Severity Schedule Name",
+        "    Controller:OutdoorAir,   !- Controller Object Type",
+        "    VAV_1_OA_Controller,     !- Controller Object Name",
+        "    5000;                    !- Enthalpy Sensor Offset {J/kg}",
+
+        "  FaultModel:TemperatureSensorOffset:ReturnAir,",
+        "    RATFault,                !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    ,                        !- Severity Schedule Name",
+        "    Controller:OutdoorAir,   !- Controller Object Type",
+        "    VAV_2_OA_Controller,     !- Controller Object Name",
+        "    -2.0;                    !- Temperature Sensor Offset {deltaC}",
+
+        "  FaultModel:EnthalpySensorOffset:ReturnAir,",
+        "    RAHFault,                !- Name",
+        "    ,                        !- Availability Schedule Name",
+        "    ,                        !- Severity Schedule Name",
+        "    Controller:OutdoorAir,   !- Controller Object Type",
+        "    VAV_2_OA_Controller,     !- Controller Object Name",
+        "    -2000;                   !- Enthalpy Sensor Offset {J/kg}",
+    });
+
+    // Process inputs
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    ScheduleManager::ProcessScheduleInput(); // read schedules
+
+    EXPECT_TRUE(FaultsManager::GetFaultsInputFlag);
+    MixedAir::GetOAControllerInputs();
+    EXPECT_FALSE(FaultsManager::GetFaultsInputFlag);
+
+    // there are two OA controller objects
+    EXPECT_EQ(MixedAir::NumOAControllers, 2);
+    // there are five economizer faults
+    EXPECT_EQ(FaultsManager::NumFaultyEconomizer, 5);
+
+    // there are three economizer faults in the 1st OA controller
+    EXPECT_EQ(MixedAir::OAController(1).NumFaultyEconomizer, 3);
+    EXPECT_EQ(MixedAir::OAController(1).EconmizerFaultNum(1), 1);
+    EXPECT_EQ(MixedAir::OAController(1).EconmizerFaultNum(2), 2);
+    EXPECT_EQ(MixedAir::OAController(1).EconmizerFaultNum(3), 3);
+
+    // there are two economizer faults in the 2nd OA controller
+    EXPECT_EQ(MixedAir::OAController(2).NumFaultyEconomizer, 2);
+    EXPECT_EQ(MixedAir::OAController(2).EconmizerFaultNum(1), 4);
+    EXPECT_EQ(MixedAir::OAController(2).EconmizerFaultNum(2), 5);
+}
```

Comment:
:+1:

Reply:
@Myoldmopar Removed the get input flag that I added. It was unnecessary.